### PR TITLE
Simplify the PR template, make it more approachable

### PR DESCRIPTION
**Pull request by aaime**
Making the PR template shorter and easier to understand.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
+- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).
 
+For core and extension modules:
 
-For all pull requests:
+- [ ] New unit tests have been added covering the changes.
+- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en) has been updated (if change is visible to end users).
+- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
+- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/projects/GEOS) (except for changes not visible to end users). 
+- [ ] Commit message(s) must be in the form ``[GEOS-XYZW] Title of the Jira ticket``.
```
Comment (tbarsballe) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: This really doesn't matter, but now that the GEOS Jira has passed 10000 tickets, does it make more sense to say `[GEOS-XYZWV]` here?

**Review by tbarsballe — APPROVED** [review]
Looks good to me. I like the consistency with the GeoTools checklist, and it appears to cover all the same material as before.

I've added a couple minor comments inline. I don't think they need to be changed, but are things that could be considered.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
+- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
+- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).
 
+For core and extension modules:
 
-For all pull requests:
+- [ ] New unit tests have been added covering the changes.
+- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en) has been updated (if change is visible to end users).
+- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
+- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/projects/GEOS) (except for changes not visible to end users).
```
Comment (bradh) [inline .github/PULL_REQUEST_TEMPLATE.md:19]:
```suggestion
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/GEOS/issues/) (except for changes that do not affect administrators or end users in any way). 
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (bradh) [inline .github/PULL_REQUEST_TEMPLATE.md:20]:
```suggestion
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+For core and extension modules:
 
-For all pull requests:
+- [ ] New unit tests have been added covering the changes.
+- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en) has been updated (if change is visible to end users).
+- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
+- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/projects/GEOS) (except for changes not visible to end users). 
+- [ ] Commit message(s) must be in the form ``[GEOS-XYZW] Title of the Jira ticket``.
+- [ ] Bug fixes and small new features are presented as a single commit.
+- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).
```
Comment (bradh) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
+- [ ] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
+- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
+- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
+- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).
 
+For core and extension modules:
 
-For all pull requests:
+- [ ] New unit tests have been added covering the changes.
+- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en) has been updated (if change is visible to end users).
```
Comment (bradh) [inline .github/PULL_REQUEST_TEMPLATE.md:17]:
```suggestion
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
```

**Review by bradh — COMMENTED** [review]
I think its a good change. Made a few suggestions, including @tbarsballe comments.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (aaime) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: The link is actually a 404. You probably meant https://osgeo-org.atlassian.net/jira/software/c/projects/GEOS/issues/

Comment (aaime) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: I'm adding it in a commit to capture the wording change and then fix the link

**Pull request merged**